File: internal/models/channel.go
package models

import (
	"time"
)

type Channel struct {
	ID        uint      `gorm:"primaryKey" json:"id"`
	Name      string    `gorm:"unique;not null" json:"name"`
	Enabled   bool      `gorm:"default:true;not null" json:"enabled"`
	Shared    bool      `gorm:"default:false;not null" json:"shared"`
	UserID    *uint     `gorm:"column:userId" json:"userId"`
	CreatedAt time.Time `gorm:"column:createdAt" json:"createdAt"`
	UpdatedAt time.Time `gorm:"column:updatedAt" json:"updatedAt"`

	// Associations
	User          *User               `gorm:"foreignKey:UserID" json:"user,omitempty"`
	Providers     []ChannelProvider   `gorm:"foreignKey:ChannelID" json:"-"`
	AllowedModels []ChannelAllowedModel `gorm:"foreignKey:ChannelID" json:"-"`
}

func (Channel) TableName() string {
	return "Channel"
}

File: web/src/types/channel.ts
// Channel types
export interface Channel {
  id: number
  name: string
  enabled: boolean
  shared: boolean
  supportsAllModels?: boolean
  userId?: number
  // Associated providers (many-to-many)
  providers?: { id: number; name: string }[]
  // Allowed models (many-to-many)
  allowedModels?: { id: number; name: string }[]
  // Legacy support
  models?: (number | { id: number; name: string })[]
  providerId?: number
  provider_ids?: number[]
  createdAt: string
  updatedAt: string
  created_at?: string
  updated_at?: string
}

export interface CreateChannelRequest {
  name: string
  enabled?: boolean
  shared?: boolean
  supportsAllModels?: boolean
}

export interface UpdateChannelRequest {
  name?: string
  enabled?: boolean
  shared?: boolean
  supportsAllModels?: boolean
}

export interface BindProvidersRequest {
  provider_ids: number[]
}

export interface BindModelsRequest {
  model_ids: number[]
}